File: frontend/src/objects/merchant/profile/MerchantWithdrawal.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  CurrencyCents,
  DisplayText,
  IsoDateTime,
  MerchantWithdrawalId,
} from '@/objects/core/SharedObjects'

export type MerchantWithdrawal = {
  id: MerchantWithdrawalId
  amountCents: CurrencyCents
  accountLabel: DisplayText
  requestedAt: IsoDateTime
}

File: backend/src/main/scala/services/merchant/objects/MerchantWithdrawal.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class MerchantWithdrawal(
    id: MerchantWithdrawalId,
    amountCents: CurrencyCents,
    accountLabel: DisplayText,
    requestedAt: IsoDateTime,
)
object MerchantWithdrawal:
  given Encoder[MerchantWithdrawal] = deriveEncoder
  given Decoder[MerchantWithdrawal] = deriveDecoder
